source/extensions/filters/http/cache/http_cache.h

```diff
@@ -69,6 +72,13 @@ class RawByteRange {
   const uint64_t last_byte_pos_;
 };
 
+class RangeRequests : Logger::Loggable<Logger::Id::cache_filter> {
```

Comment:
It seems odd to introduce a new class just to hold this one function. What do you think of putting it in http_cache_utils.h?

Reply:
I added the class so we could use `ENVOY_LOG`, would it be better if I move it to `HttpCacheUtils` and add logging to that?

Reply:
RawByteRange and AdjustedByteRange are part of HttpCache's interface, so they should stay (or possibly move to their own dedicated headers--I could go either way). parseRanges is part of HttpCache's implementation, and that's exactly what HttpCacheUtils is for.

Reply:
Since `parseRanges` uses `RawByteRange`, if I just move that function we will have a circular dependency, so it doesn't work.
The options I can think of:
1. move `parseRanges` and `RawByteRange` (+`AdjustedByteRange` for consistency?) into a new file;
2. remove logging so we can keep `parseRanges` just as function inside `http_cache`;
3. keep the extra class with only this function inside.

Do you have any other ideas?
If not, which of these would you prefer? (Out of these, I'm more inclined with 1)

Reply:
Just leave the code as it is; we've spent more time than this one was worth.